src/app/workflow/version-modal/version-modal.service.ts

```diff
@@ -64,28 +65,28 @@ export class VersionModalService {
     workflowVersion: WorkflowVersion,
     originalTestParameterFilePaths,
     newTestParameterFiles,
-    workflowMode: String
+    workflowMode: Workflow.ModeEnum
   ) {
     const message1 = 'Saving workflow version';
-    const message2 = 'Modifying test parameter files';
     const workflowId = this.workflowQuery.getActive().id;
     let toRefresh = false;
     // Checks if the workflow path was changed
     if (originalVersion.workflow_path !== workflowVersion.workflow_path) {
       toRefresh = true;
     }
     this.alertService.start(message1);
-    if (workflowMode !== 'HOSTED') {
-      this.workflowsService.updateWorkflowVersion(workflowId, [workflowVersion]).subscribe(
-        (response) => {
-          const updatedWorkflow: BioWorkflow | Service | AppTool = { ...workflow, workflowVersions: response };
-          this.workflowService.setWorkflow(updatedWorkflow);
-          this.alertService.start(message2);
+    this.workflowsService.updateWorkflowVersion(workflowId, [workflowVersion]).subscribe(
+      (response) => {
+        const updatedWorkflow: BioWorkflow | Service | AppTool = { ...workflow, workflowVersions: response };
+        this.workflowService.setWorkflow(updatedWorkflow);
+        if (workflowMode === ModeEnum.HOSTED) {
+          this.alertService.detailedSuccess();
+          this.matDialog.closeAll();
+        } else {
           this.modifyTestParameterFiles(workflowVersion, originalTestParameterFilePaths, newTestParameterFiles).subscribe(
-            (success) => {
+            (files) => {
               // Checks if there was a test parameter file added/removed
-              // The this.modifyTestParameterFiles function returns a {} observable if nothing was done, this is a way of checking for it
-              if (!(Object.keys(success).length === 0 && success.constructor === Object)) {
+              if (files !== null) {
```

Comment:
If I'm reading the code right, in all cases, even when `modifyTestParameterFiles` returns the value of `observableOf(null)`, `files` is an `Observable` object, so this conditional will always evaluate to true.  But that's ok, I propose we always refresh the version after we try to update it, because the overhead is low, and we'll avoid the case where a bug prevents us from seeing the updated version.

Reply:
`files` is `SourceFile[] | null` because we're in the subscribe. "Proof" of a sorts is this WebStorm screenshot:

![Screen Shot 2022-08-23 at 3 56 30 PM](https://user-images.githubusercontent.com/1049340/186280462-96157e30-ebef-4d72-914d-54b4b250a8ec.png)

Plus I had run it in the debugger as well, and was able to verify that.

I think the idea is that just toggling `hidden` shouldn't refresh the whole workflow/version, as that's sort of an expensive operation.

Reply:
Yes, you are totally correct on all counts.  Disregard.